python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -191,33 +201,82 @@ def _load_multiple_inputs(self, context: InputContext) -> Dict[str, Any]:
             else False
         )
 
-        objs: Dict[str, Any] = {}
-        paths = self._get_paths_for_partitions(context)
-        backcompat_paths = self._get_multipartition_backcompat_paths(context)
+        paths = self._get_paths_for_partitions(context)  # paths for normal partitions
+        backcompat_paths = self._get_multipartition_backcompat_paths(
+            context
+        )  # paths for multipartitions
 
         context.log.debug(f"Loading {len(paths)} partitions...")
 
-        for partition_key, path in paths.items():
-            context.log.debug(f"Loading partition from {path} using {self.__class__.__name__}")
+        def load_partition(partition_key: Union[str, MultiPartitionKey]):
```

Comment:
With all the additions here, `_load_multiple_inputs` is starting to get a little large and unwieldy. I think moving `load_partition` to be its own method on `UPathIOManager` would make this easier to follow.